src/rocker/core.py

```diff
@@ -251,12 +251,24 @@ def docker_build(docker_client = None, output_callback = None, **kwargs):
         print("no more output and success not detected")
         return None
 
-def docker_remove_image(image_id, docker_client = None, output_callback = None, **kwargs):
+def docker_remove_image(
+        image_id,
+        docker_client = None,
+        output_callback = None,
+        fail_on_error = False,
+        force = False,
+        **kwargs):
 
     if not docker_client:
         docker_client = get_docker_client()
 
-    docker_client.remove_image(image_id)
+    try:
+        docker_client.remove_image(image_id, force=force)
+    except docker.errors.APIError as ex:
+        ## removing the image can fail if there's child images
```

Comment:
It seems a little weird not to print or log a message when catching this error. The end-user has no way of knowing whether the image was cleaned up or not, especially with the default behavior of not failing on error.

Reply:
This will return a True/False and then it warns the user based on the results: https://github.com/osrf/rocker/blob/4a6e19171765e417d211ec9a139a66987bca5645/src/rocker/core.py#L437

And the default is False so it's only suppressed if the developer asks for it to be suppressed.